File: python/algorithms/easy/miniMaxSum.py
"""
Author: Rithie Natan Carvalhaes Prado
Date: 2026-06-13
Last modified: 2026-06-13
Description: Mini-Max Sum - HackerRank
Test description: https://www.hackerrank.com/challenges/mini-max-sum/problem
"""
#---------- functions ----------
def miniMaxSum(arr: list[int]) -> None:
    """
    Calculate the minimum and maximum sum of an array.

    Parameters:
        arr (list[int]): An array of integers.
    """

    arr.sort()

    min_sum = sum(arr[:-1])
    max_sum = sum(arr[1:])

    print(min_sum, max_sum)

File: java/algorithms/easy/miniMaxSum.java
/**
 * @author Rithie Natan Carvalhaes Prado
 * @date 2026-06-13
 * Last modified: 2026-06-13
 * @description Plus Minus - HackerRank
 * @see {@link https://www.hackerrank.com/challenges/plus-minus/problem}
 */
//---------- Imports ----------
import java.util.List;


//---------- solution ----------
class Result
{
    /**
     * Calculate the minimum and maximum sum of an array.
     * @param {List<Integer>} arr
     */
    public static void miniMaxSum(List<Integer> arr)
    {
        arr.sort(Integer::compareTo);

        long min = arr.stream().limit(arr.size() - 1).mapToLong(Integer::longValue).sum(); 
        long max = arr.stream().skip(1).mapToLong(Integer::longValue).sum();

        System.out.println(min + " " + max);
    }//end miniMaxSum()

}//end class